.github/workflows/tests.yaml

```
@@ -0,0 +1,76 @@
+name: Tests for MORS
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.10', '3.11', '3.12']
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup system
+        run: |
+          sudo apt-get update
+          sudo apt-get install libnetcdff-dev netcdf-bin
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - uses: actions/cache@v3
+        id: cache-virtualenv
+        with:
+          path: ${{ env.pythonLocation }}
+          key: ${{ env.pythonLocation }}-${{ hashFiles('pyproject.toml') }}
+
+      - name: Build MORS
+        if: steps.cache-virtualenv.outputs.cache-hit != 'true'
+        run: |
+          python -m pip install -e .[develop]
+
+      - uses: actions/cache@v3
+        id: cache-fwl-data
+        with:
+          path: /home/runner/work/fwl_data
+          key: fwl-data-1
+
+      - name: Test with pytest
+        run: |
+          export FWL_DATA="/home/runner/work/fwl_data"
+          coverage run -m pytest
+
+      - name: Report coverage
+        run: |
+          coverage json
+          export TOTAL=$(python -c "import json;print(json.load(open('coverage.json'))['totals']['percent_covered_display'])")
+          echo "total=$TOTAL" >> $GITHUB_ENV
+          echo "### Total coverage: ${TOTAL}%" >> $GITHUB_STEP_SUMMARY
+          echo $'\n```' >> $GITHUB_STEP_SUMMARY
+          coverage report >> $GITHUB_STEP_SUMMARY
+          echo $'\n```' >> $GITHUB_STEP_SUMMARY
+
+      - name: Make coverage badge
+        if: ${{ github.ref == 'refs/heads/master' && matrix.python-version == '3.10' }}
+        uses: schneegans/dynamic-badges-action@v1.7.0
+        with:
+          auth: ${{ secrets.GIST_TOKEN }}
+          gistID: 99391a66bb9229771504c3a4db611d05
```

The way this works is that you make a gist, does not matter what is in there, and copy the gist ID. The action will update the badge if succesful, this ID points to JANUS: https://gist.github.com/stefsmeets/99391a66bb9229771504c3a4db611d05

Then generate a new token scoped to modifying gists, named GIST_TOKEN: https://github.com/settings/tokens

Then you update the GIST_TOKEN in https://github.com/FormingWorlds/MORS/settings/secrets/actions under Repository secrets. Ask @timlichtenberg to give you permissions (I don't have them).



```suggestion
          gistID: your own
```

---

Gist is created and ID updated in README and tests.yaml.
I still need to create associated token and store it in the repo but this is outside this PR.